dev-docs/RFCs/vNext/icon-manager-rfc.md

```diff
@@ -0,0 +1,55 @@
+# RFC: Icon Manager 
+
+* **Authors**: Xintong Xia 
+* **Date**: December 2018
+* **Status**: Draft 
+
+## Abstract
+
+This RFC proposes add `IconManager` class to help manage `iconAtlas` and `iconMapping` used in `IconLayer`.
+
+## Motivation 
+
+Currently `IconLayer` requires user pass `iconAtlas` and `iconMapping` when constructing a `IconLayer` instance. 
+But there are some cases that `iconAtlas` can not be decided beforehand, requiring a way to dynamically update. 
+For example, the icons may need to be fetched from a remote url.
+
+Refer [IconLayer](/docs/layers/icon-layer.md) for details.
+ 
+## Proposal: Add **IconManager** class
+
+`IconManager` is responsible for 
+
+- Calculating `iconMapping` to generate an `iconAtlas` with all the required icons based on the size.
+- Constructing a `Texture2D` object sufficient to fit in the `iconAtlas`. 
+- Loading all the icons and update texture.
+
+### Constructor
+
+```js
+const IconManager = new IconManager(gl, {
+  data,
+  getIcon,
+  onTextureUpdate,
+  padding,
+  maxCanvasWidth
+})
+```
+
+### Parameters
+
+- **data(Object)**: data feed into IconLayer
+- **getIcon(Function)**: accessor to get icon from each data point
```

Comment:
What does "getIcon" function return? Can users pass an image path, url, or an "Image" object? Does it require all the icons have the same dimension?  If it's async loading the icon, maybe users can specify a placeholder icon when loading?